packages/dashboard/src/PanelManager.ts

```diff
@@ -322,12 +355,15 @@ class PanelManager {
   }
 
   addClosedPanel(glContainer: Container): void {
+    const { root } = this.layout;
     const config = LayoutUtils.getComponentConfigFromContainer(glContainer);
     if (config && isReactComponentConfig(config)) {
       const dehydratedConfig = this.dehydrateComponent(
         config.component,
         config
       );
+      (dehydratedConfig as ClosedPanel).parentStackId =
+        LayoutUtils.getStackForConfig(root, config)?.config.id;
       if (dehydratedConfig != null) {
         this.closed.push(dehydratedConfig);
       }
```

Comment:
We shouldn't need the `null` check here, the `dehydratedConfig` cannot be `null`.